src/XPath2/AST/PathExprNode.cs

```diff
@@ -130,19 +134,35 @@ private bool IsOrderedSet()
 
         public override object Execute(IContextProvider provider, object[] dataPool)
         {
+            IContextProvider localProvider;
```

Comment:
There are some existing checks in `provider != null`, does that also mean that `localProvider` can also be **null** ?

Reply:
Yes. localProvider has the same value than provider if we are not in the case of an "_evaluateFromRoot".
The only change is when we are in the case, so in this case localProvider is not null and is initialised to the Root node.

Reply:
Should in case the localProvider be declared as nullable?

Reply:
Ok I see:

public override object Execute(IContextProvider provider, object[] dataPool)

provider is not declared as nullable, but it can effectly be nullable, peharps best to change the signature to specify provider is nullable ? And also signature for others methods

I correct just the localProvider to be nullable and push the commit